### DISCUSSION TITLE
Converting Snowflake, RedShift, Big Query, Trino, ClickHouse, Druid SQL to StarRocks SQL (WIP)

### DISCUSSION THREAD

**Discussion by alberttwong** [StarRocks/starrocks#25016] [How-tos, Tutorials, Best Practices and Architecture Articles]
### Preferred

This tutorial is to show how to quickly build a Streaming ELT job from MySQL to StarRocks using Flink CDC 3.0，including the feature of sync all table of one database, schema change evolution and sync sharding tables into one table.

https://ververica.github.io/flink-cdc-connectors/master/content/quickstart/mysql-starrocks-pipeline-tutorial.html

### General advice for SQL converting

Check out this project - https://github.com/tobymao/sqlglot

### Trino Read Query Support in StarRocks

Trino: As of 3.1 we support the Trino dialect for query. 

### Converting a Snowflake data load into a StarRocks data load

https://github.com/StarRocks/starrocks/discussions/24505

### Product Feature substitution from StarRocks to Snowflake, RedShift, BigQuery

https://github.com/StarRocks/starrocks/discussions/23805

<img referrerpolicy="no-referrer-when-downgrade" src="https://static.scarf.sh/a.png?x-pxid=6ed38bd5-ef12-46bd-949c-f5d6ac7031c2" />

**Comment by drshn** [comment]
Would love to see progress on this task as it makes migration easier to Starrocks. Thanks :)

**Reply by alberttwong** [reply]
We built a Trino dialect recently so that will allow you to basically migrate without SQL changes.   I think we have planned work for others.   What are priorities for you?
